spatialmath/geometry.go

```diff
@@ -15,12 +15,12 @@ type Geometry interface {
 	AlmostEqual(Geometry) bool
 	Transform(Pose) Geometry
 	ToProtobuf() *commonpb.Geometry
-	CollidesWith(Geometry) (bool, error)
+	CollidesWith(Geometry, float64) (bool, error)
 	// If DistanceFrom is negative, it represents the penetration depth of the two geometries, which are in collision.
 	// Penetration depth magnitude is defined as the minimum translation which would result in the geometries not colliding.
 	// For certain entity pairs (box-box) this may be a conservative estimate of separation distance rather than exact.
-	DistanceFrom(Geometry) (float64, error)
-	EncompassedBy(Geometry) (bool, error)
+	DistanceFrom(Geometry, float64) (float64, error)
+	EncompassedBy(Geometry, float64) (bool, error)
```

Comment:
Prior to this work being done, the variable was *called* "collision buffer", because that's what it was doing, but the intention behind it was more to account for minor floating-point variations that would make precisely contacting geometries not behave deterministically on different architectures.

I would agree that it is odd to accept a buffer on "DistanceFrom". I would suggest that we persist the 1e-8 as a private const in `spatialmath` and use if for these two functions.